packages/contentful--create-contentful-app/lib/index.js

```diff
@@ -2,54 +2,56 @@
 
 /* eslint-disable no-console, no-process-exit */
 
-const chalk = require('chalk');
-const spawn = require('cross-spawn');
-const path = require('path');
-const tildify = require('tildify');
-const { createAppDefinition } = require('@contentful/app-scripts');
-const { version } = require('../package.json');
+import { createAppDefinition } from '@contentful/app-scripts';
+import chalk from 'chalk';
+import degit from 'degit';
+import { readFileSync, writeFileSync } from 'fs';
+import { basename, resolve } from 'path';
+import tildify from 'tildify';
+import { exec, rmIfExists } from './utils.js';
+import os from 'os';
 
 const command = process.argv[2];
-const appFolder = process.argv[3];
 
 const localCommand = '@contentful/create-contentful-app';
 const mainCommand = `npx ${localCommand}`;
 
-function getTemplate() {
-  const templatePkg = '@contentful/cra-template-create-contentful-app';
-
-  if (!process.env.USE_LINKED_TEMPLATE) {
-    return `${templatePkg}@${version}`;
-  }
-
-  const linkedTemplatePath = path.relative(
-    process.cwd(),
-    path.dirname(require.resolve(templatePkg))
-  );
-
-  console.log();
-  console.log(chalk.dim('> Using linked template at %s'), chalk.blueBright(linkedTemplatePath));
-  console.log();
-
-  return `file:${linkedTemplatePath}`;
-}
-
-function onSuccess(folder) {
+function successMessage(folder) {
   console.log(`
-${chalk.cyan('Success!')} Created a new Contentful app in ${chalk.bold(
-    tildify(path.resolve(process.cwd(), folder))
-  )}.
+${chalk.cyan('Success!')} Created a new Contentful app in ${chalk.bold(tildify(folder))}.
 
 We suggest that you begin by running:
 
     ${chalk.cyan(`cd ${folder}`)}
     ${chalk.cyan(`${mainCommand} create-definition`)}
   `);
+}
+
+function updatePackageName(appFolder) {
+  const packageJsonPath = resolve(appFolder, 'package.json');
+  const packageJson = JSON.parse(readFileSync(packageJsonPath, { encoding: 'utf-8' }));
+  packageJson.name = basename(appFolder);
+  writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2) + os.EOL);
+}
+
+async function cloneTemplate(name, destination) {
+  const d = degit(`contentful/apps/templates/${name}`, { mode: 'tar' });
 
-  process.exit();
+  try {
+    await d.clone(destination);
+  } catch (e) {
+    let message = 'Error creating app';
+    if (e.code === 'DEST_NOT_EMPTY') {
+      message = 'destination directory is not empty';
+    }
+    throw new Error(message);
+  }
 }
 
-function initProject() {
+async function initProject() {
+  const appFolder = process.argv[3];
```

Comment:
should we maybe validate if this is a correct path-like string?

Reply:
What is a path-like string?

Should `path.resolve(...)` handle all these cases and just return a proper absolute path?

Reply:
I don't think we get any validation there

**Node REPL**
```
path.resolve(process.cwd(), '>>><>?||')
'$ABSOLUTE_PATH/>>><>?||'
```

But on Windows, for example, `\ / : * ? " < > |` are invalid chars for filenames. Maybe it's handled in the windows version of Node (and therefore `path`)